internal-webapps/org.wso2.bfsi.authentication.endpoint/src/main/java/org/wso2/bfsi/authentication/endpoint/BFSIConsentConfirmServlet.java

```diff
@@ -0,0 +1,173 @@
+                metadata.putAll(updatedMetadata);
+            }
+
+            Map<String, Object> updatedConsentData = bfsiAuthServletTK.updateConsentData(request);
+            if (updatedConsentData != null) {
+                updatedConsentData.forEach(consentData::put);
+            }
+        }
+
+        consentData.put(Constants.METADATA, metadata);
+
+        String redirectURL = persistConsentData(
+                consentData, request.getParameter(Constants.SESSION_DATA_KEY_CONSENT), getServletContext());
+
+        // Invoke authorize flow
+        if (redirectURL != null) {
+            response.sendRedirect(redirectURL);
+
+        } else {
+            session.invalidate();
+            response.sendRedirect("retry.do?status=Error&statusMsg=Error while persisting consent");
+        }
+
+    }
+
+    @Generated(message = "Contains the tested code of HTTPClient")
+    String persistConsentData(JSONObject consentData, String sessionDataKey, ServletContext servletContext) {
+
+        String persistenceBaseURL = servletContext.getInitParameter(Constants.PERSISTENCE_BASE_URL);
+        String persistenceUrl = persistenceBaseURL + Constants.SLASH + sessionDataKey;
+
+        try (CloseableHttpClient client = HttpClientBuilder.create().build()) {
+            HttpPatch dataRequest = new HttpPatch(persistenceUrl);
+            dataRequest.addHeader(Constants.ACCEPT, Constants.JSON);
+            dataRequest.addHeader(Constants.AUTHORIZATION, Constants.BASIC +
+                    BFSIConsentServlet.getConsentApiCredentials());
+            StringEntity body = new StringEntity(consentData.toString(), ContentType.APPLICATION_JSON);
+            dataRequest.setEntity(body);
+            HttpResponse dataResponse = client.execute(dataRequest);
+
+            if (dataResponse.getStatusLine().getStatusCode() == HttpURLConnection.HTTP_MOVED_TEMP &&
+                    dataResponse.getLastHeader(Constants.LOCATION) != null) {
+                return dataResponse.getLastHeader(Constants.LOCATION).getValue();
+            } else {
+                String retrievalResponse = IOUtils.toString(dataResponse.getEntity().getContent(),
+                        String.valueOf(StandardCharsets.UTF_8));
+                JSONObject data = new JSONObject(retrievalResponse);
+                String errorResponse = AuthenticationUtils.getErrorResponseForRedirectURL(data);
+                if (data.has(Constants.REDIRECT_URI) && StringUtils.isNotEmpty(errorResponse)) {
+                    URI errorURI = new URI(data.get(Constants.REDIRECT_URI).toString().concat(errorResponse));
+                    return errorURI.toString();
+                } else {
+                    return null;
+                }
+            }
+        } catch (IOException | JSONException | URISyntaxException e) {
+            log.error("Exception while calling persistence endpoint", e);
+            return null;
+        }
+    }
+    /**
+     * Retrieve the config.
```

Comment:
Shall we make the comment bit more descriptive?

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/8/commits/7c13892e0735c7d32a51565d4369ab1e9eddbc7e